### ISSUE TITLE
[wxwidgets] Build error

### ISSUE THREAD

**Issue by wieniawski-lai** [microsoft/vcpkg#30060]
**Host Environment**

- OS: win10
- Compiler: VS2022, C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.30.30705/bin/Hostx64/x86/cl.exe

**To Reproduce**

`./vcpkg install wxWidgets`
before that, I also tried
```
git pull
vcpkg update
```
to  make sure all my data is the newest

**Failure logs**
![image](https://user-images.githubusercontent.com/70927596/223318547-43d84b77-f41c-4490-8b13-715101b49ddb.png)

- Cut and paste the appropriate build messages from the console output.
- Attach ***every failure log file*** mentioned in the console output.
log files can be found here:
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/10905318/install-x86-windows-dbg-out.log)

[install-x86-windows-dbg-out-1.log](https://github.com/microsoft/vcpkg/files/10905315/install-x86-windows-dbg-out-1.log)

Pls help to see what is the matter.

thx.

**Comment by MonicaLiu0311** [comment]
Win 10
VS 2022

```
PS E:\vcpkg03> ./vcpkg version
vcpkg package management program version 2023-03-01-538539c4e364dd6f14f5e36f6704144254845159

See LICENSE.txt for license information.
```

```
PS E:\vcpkg03> ./vcpkg install wxWidgets
warning: Starting with the September 2023 release, the default triplet for vcpkg libraries will change from x86-windows to the detected host triplet (x64-windows). To resolve this message, add --triplet x86-windows to keep the same behavior.
Computing installation plan...
The following packages are already installed:
    wxwidgets[core,debug-support,sound]:x86-windows -> 3.2.2.1
wxwidgets:x86-windows is already installed
Restored 0 package(s) from C:\Users\v-liumonica\AppData\Local\vcpkg\archives in 132.3 us. Use --debug to see more details.
Total install time: 341.7 us
The package wxwidgets provides CMake targets:

    find_package(wxWidgets CONFIG REQUIRED)
    target_link_libraries(main PRIVATE wx::core wx::base)
```

Sorry I can't reproduce your issue, the local installation was successful.

**Comment by dg0yt** [comment]
@MonicaLiu0311 You should actually try to build (fresh install dir, no binary cache), not just show that `wxwidgets:x86-windows is already installed`.

**Comment by MonicaLiu0311** [comment]
> @MonicaLiu0311 You should actually try to build (fresh install dir, no binary cache), not just show that wxwidgets:x86-windows is already installed.

Yes, that's what I did, because the installation log is too long, so I deliberately took the log of repeated installation, vcpkg03 is a newly cloned library.

```
PS E:\vcpkg03> ./vcpkg install  wxWidgets --binarysource=clear
warning: Starting with the September 2023 release, the default triplet for vcpkg libraries will change from x86-windows to the detected host triplet (x64-windows). To resolve this message, add --triplet x86-windows to keep the same behavior.
Computing installation plan...
The following packages will be built and installed:
    wxwidgets[core,debug-support,sound]:x86-windows -> 3.2.2.1
Detecting compiler hash for triplet x86-windows...
Installing 1/1 wxwidgets:x86-windows...
Building wxwidgets[core,debug-support,sound]:x86-windows...
-- Using cached wxWidgets-wxWidgets-v3.2.2.1.tar.gz.
-- Cleaning sources at E:/vcpkg03/buildtrees/wxwidgets/src/v3.2.2.1-8c75496f16.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source E:/vcpkg03/downloads/wxWidgets-wxWidgets-v3.2.2.1.tar.gz
-- Applying patch install-layout.patch
-- Applying patch relocatable-wx-config.patch
-- Applying patch nanosvg-ext-depend.patch
-- Applying patch fix-libs-export.patch
-- Applying patch fix-pcre2.patch
-- Applying patch gtk3-link-libraries.patch
-- Applying patch sdl2.patch
-- Applying patch fix_include.patch
-- Using source at E:/vcpkg03/buildtrees/wxwidgets/src/v3.2.2.1-8c75496f16.clean
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at E:/vcpkg03/downloads/tools/msys2/9a1ec3f33446b195
-- Found external ninja('1.11.0').
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Installing: E:/vcpkg03/packages/wxwidgets_x86-windows/lib/mswud/wx/setup.h
-- Installing: E:/vcpkg03/packages/wxwidgets_x86-windows/share/wxwidgets/usage
-- Installing: E:/vcpkg03/packages/wxwidgets_x86-windows/share/wxwidgets/copyright
-- Performing post-build validation
Elapsed time to handle wxwidgets:x86-windows: 4.638 min
Total install time: 4.863 min
The package wxwidgets provides CMake targets:

    find_package(wxWidgets CONFIG REQUIRED)
    target_link_libraries(main PRIVATE wx::core wx::base)
```

**Comment by MonicaLiu0311** [comment]
> Please ensure your sysytem has sufficient memory.

@wieniawski-lai 
I found this, can you confirm it?

**Comment by dg0yt** [comment]
~~~
LINK : fatal error LNK1104: cannot open file 'lib\wxmsw32ud_core_vc_custom.dll'
~~~
Maybe access blocked by anti-virus software.

**Comment by wieniawski-lai** [comment]
> lib\wxmsw32ud_core_vc_custom.dll

no,

![image](https://user-images.githubusercontent.com/70927596/223637667-c707ba9d-a0c1-45ee-9477-e02a5e740c2a.png)

Indeed I can not find such dll: 'lib\wxmsw32ud_core_vc_custom.dll'

I guess it should be compiled output, but I donot why it missed.

**Comment by dg0yt** [comment]
This fits very well in the failed-due-to-anti-virus pattern.

**Comment by wieniawski-lai** [comment]
> This fits very well in the failed-due-to-anti-virus pattern.

the only anti-virus software in my computer is windows defender....

**Comment by wieniawski-lai** [comment]
> > @MonicaLiu0311 You should actually try to build (fresh install dir, no binary cache), not just show that wxwidgets:x86-windows is already installed.
> 
> Yes, that's what I did, because the installation log is too long, so I deliberately took the log of repeated installation, vcpkg03 is a newly cloned library.
> 
> ```
> PS E:\vcpkg03> ./vcpkg install  wxWidgets --binarysource=clear
> warning: Starting with the September 2023 release, the default triplet for vcpkg libraries will change from x86-windows to the detected host triplet (x64-windows). To resolve this message, add --triplet x86-windows to keep the same behavior.
> Computing installation plan...
> The following packages will be built and installed:
>     wxwidgets[core,debug-support,sound]:x86-windows -> 3.2.2.1
> Detecting compiler hash for triplet x86-windows...
> Installing 1/1 wxwidgets:x86-windows...
> Building wxwidgets[core,debug-support,sound]:x86-windows...
> -- Using cached wxWidgets-wxWidgets-v3.2.2.1.tar.gz.
> -- Cleaning sources at E:/vcpkg03/buildtrees/wxwidgets/src/v3.2.2.1-8c75496f16.clean. Use --editable to skip cleaning for the packages you specify.
> -- Extracting source E:/vcpkg03/downloads/wxWidgets-wxWidgets-v3.2.2.1.tar.gz
> -- Applying patch install-layout.patch
> -- Applying patch relocatable-wx-config.patch
> -- Applying patch nanosvg-ext-depend.patch
> -- Applying patch fix-libs-export.patch
> -- Applying patch fix-pcre2.patch
> -- Applying patch gtk3-link-libraries.patch
> -- Applying patch sdl2.patch
> -- Applying patch fix_include.patch
> -- Using source at E:/vcpkg03/buildtrees/wxwidgets/src/v3.2.2.1-8c75496f16.clean
> -- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
> -- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
> -- Using msys root at E:/vcpkg03/downloads/tools/msys2/9a1ec3f33446b195
> -- Found external ninja('1.11.0').
> -- Configuring x86-windows
> -- Building x86-windows-dbg
> -- Building x86-windows-rel
> -- Installing: E:/vcpkg03/packages/wxwidgets_x86-windows/lib/mswud/wx/setup.h
> -- Installing: E:/vcpkg03/packages/wxwidgets_x86-windows/share/wxwidgets/usage
> -- Installing: E:/vcpkg03/packages/wxwidgets_x86-windows/share/wxwidgets/copyright
> -- Performing post-build validation
> Elapsed time to handle wxwidgets:x86-windows: 4.638 min
> Total install time: 4.863 min
> The package wxwidgets provides CMake targets:
> 
>     find_package(wxWidgets CONFIG REQUIRED)
>     target_link_libraries(main PRIVATE wx::core wx::base)
> ```

I found that you are using ninja v1.11.0. And my version is 1.10.2.

However, I installed ninja outside of vcpkg, whose version is 1.11.1.

The version 1.10.2 is downloaded and compiled by vcpkg. I donot why even I have installed ninja before, vcpkg will install a lower version itself.

Similar things happens in 7z, which I installed in my computer before and seperately. But vcpkg would download again and build a seperate one...

I donot know how to control vcpkg not to build the software which I have already isntalled... which is another topic...

**Comment by wieniawski-lai** [comment]
en, sth. new happens.

I turned off my windows-defender, and close virtual box, to make sure I have sufficient memory. And rebuild with the same command. And I monitored my memory, which is less than 60% in building.

At this time:
- debug version is build succesfully.
- release version failed.

here are the logs:
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/10918330/install-x86-windows-dbg-out.log)
[install-x86-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/10918329/install-x86-windows-rel-out.log)

the release version's error is:
mt.exe : general error c101008d: Failed to write the updated manifest to the resource of file "lib\wxmsw32u_qa_vc_custom.dll". 

however, in the directory:
![image](https://user-images.githubusercontent.com/70927596/223652930-48260f00-3688-4ba2-bf30-b0b2844e1b33.png)
